### PR TITLE
ID-1086 Fix JSON dependency vulnerability

### PR DESCRIPTION
**Pull request by tlangs**
Ticket:  https://broadworkbench.atlassian.net/browse/ID-1086

Looks like this _actually_ does what I intended to do, which was to make sure the Sam Client doesn't ship with a vulnerable transitive dependency

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: codegen_java/templates/libraries/okhttp-gson/build.sbt.mustache [hunk 1]
```diff
@@ -39,7 +39,8 @@ lazy val root = (project in file(".")).
       "com.google.code.findbugs" % "jsr305" % "3.0.2" % "compile",
       "org.junit.jupiter" % "junit-jupiter-api" % "5.9.1" % "test",
       "com.novocode" % "junit-interface" % "0.10" % "test",
-      "org.mockito" % "mockito-core" % "3.12.4" % "test"
+      "org.mockito" % "mockito-core" % "3.12.4" % "test",
+      "org.json" % "json" % "20240205"
     ),
     dependencyOverrides += "org.json" % "json" % "20240205") ++ publishSettings:_*
```
Comment (dvoet) [inline codegen_java/templates/libraries/okhttp-gson/build.sbt.mustache]: do you still need the dependencyOverrides?

Comment (tlangs) [reply codegen_java/templates/libraries/okhttp-gson/build.sbt.mustache]: Lemme try without it.

Comment (tlangs) [reply codegen_java/templates/libraries/okhttp-gson/build.sbt.mustache]: Looks like we don't. I'm wondering why `dependencyOverrides` didn't work, but I'm not so interested that I'm going to investigate it...

**Review by dvoet — APPROVED** [review]

File: codegen_java/templates/libraries/okhttp-gson/build.sbt.mustache (hunk 1 above)
Comment (dvoet) [reply codegen_java/templates/libraries/okhttp-gson/build.sbt.mustache]: it is probably an override for the local build but doesn't affect the pom, but I am just guessing

**Review by okotsopoulos — APPROVED** [review]

**Pull request merged**